### PR TITLE
fix: prevent frozen InputSearch after text editing

### PR DESCRIPTION
**Pull request by wp-aberg**
## What I did

This PR slightly refactors input search to better respond to keyboard interaction and edits to selected text, as well as adding an interaction test story to validate

Steps to test

1. Start searching
2. Key down to one of the autocomplete suggestions
3. Try making a change to the autocomplete result in the search bar by backspacing or typing more letters
4. Changes **are** made as expected
5. Clear the search bar by clicking the x icon, it clears as expected

STRY-679

### CODE REVIEW

File: packages/kit/src/input-search/InputSearchInput.tsx [hunk 1]
```diff
@@ -44,6 +44,7 @@ export const InputSearchInput = React.forwardRef<
     const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
       if (rest.onChange) rest.onChange(event);
       if (inputProps.onChange) inputProps.onChange(event);
+      console.log("InputSearchInput handleChange", event.target.value);
```
Comment (tamstephanie) [inline packages/kit/src/input-search/InputSearchInput.tsx:47]: Reminder to remove

**Review by tamstephanie — APPROVED** [review]
LGTM!

File: packages/kit/src/input-search/InputSearchInput.tsx (hunk 1 above)
Comment (wp-aberg) [reply packages/kit/src/input-search/InputSearchInput.tsx]: ty

File: packages/kit/src/input-search/play.stories.tsx [hunk 2]
```diff
@@ -292,3 +292,23 @@ export const Interactions = {
...
+    });
+    await userEvent.keyboard("[ArrowDown]");
+    await userEvent.keyboard("[ArrowDown]");
+    await userEvent.keyboard("[ArrowDown]");
+    await expect(input).toHaveDisplayValue("Orange");
+    await userEvent.keyboard("[Backspace]");
+    await expect(input).toHaveDisplayValue("Orang");
+    const clearButton = await screen.findByText("Clear");
+    await userEvent.click(clearButton);
+    await expect(input).toHaveDisplayValue("");
```
Comment (val-l-hosler) [inline packages/kit/src/input-search/play.stories.tsx:312]: Ooo very cool!

**Review by val-l-hosler — APPROVED** [review]

**Pull request merged**